src/main/groovy/com/github/jrubygradle/jar/JRubyJarConfigurator.groovy

```diff
@@ -59,33 +72,51 @@ class JRubyJarConfigurator {
      */
     @Incubating
     void mainClass(final String className) {
-        jar().with {
+        if (this.hasMainClassName) {
+            throw new StopExecutionException('mainClass can be set only once')
+        }
+        if (className == null) {
+            throw new StopExecutionException('mainClass can be null')
+        }
+        archive.with {
             manifest {
                 attributes 'Main-Class': className
             }
         }
-        jar().with archive.project.copySpec {
-            from {
-                archive.project.configurations.jrubyJar.collect {
-                    archive.project.zipTree( it )
-                }
-            }
-            include '**'
-            exclude 'META-INF/MANIFEST.MF'
-            // some pom.xml are readonly which creates problems
-            // with zipTree on second run
-            exclude 'META-INF/maven/**/pom.xml'
-        }
+        this.hasMainClassName = true
     }
 
     @Incubating
-    void initScript(final String scriptName) {
-        def script = archive.project.file(scriptName)
-        jar().with archive.project.copySpec {
-            from(script.parent)
-            include script.name
-            rename { 'jar-bootstrap.rb' }
+    void initScript(final Object scriptName) {
+        if (this.scriptName != null) {
```

Comment:
I am not sure about this:
* Preventing the user from setting the scriptName twice is not very gradlesque (there are reasons for having to do this in complex builds).
* I think the evaluation is happening too early. the conversion should probably only happen when the task is executed.